allure-robotframework/src/listener.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import absolute_import
+from allure_commons.model2 import TestResultContainer, TestResult, TestStepResult, TestAfterResult, TestBeforeResult, \
+    StatusDetails
+from allure_commons.reporter import AllureReporter
+from allure_commons.utils import now, uuid4
+from allure_commons.logger import AllureFileLogger
+from allure_commons.types import AttachmentType
+from allure_commons import plugin_manager
+from robot.libraries.BuiltIn import BuiltIn
+from allure_robotframework.constants import RobotKeywordType, RobotLogLevel
+from allure_robotframework import utils
+import os
+
+
+# noinspection PyPep8Naming
+class allure_robotframework(object):
+    ROBOT_LISTENER_API_VERSION = 2
+    DEFAULT_OUTPUT_PATH = os.path.join('output', 'allure')
+    LOG_MESSAGE_FORMAT = '{full_message}<p><b>[{level}]</b> {message}</p>'
+    FAIL_MESSAGE_FORMAT = '{full_message}<p style="color: red"><b>[{level}]</b> {message}</p>'
+
+    def __init__(self, logger_path=DEFAULT_OUTPUT_PATH):
+        self.reporter = AllureReporter()
+        self.logger = AllureFileLogger(logger_path)
+        self.stack = []
```

Comment:
This stack basically duplicates OrderedDict inside AllureReporter.
I think we can simply make a item_type argument of AllureReporter.get_last_item optional, so it would return the last element inserted in the default case, this way you will get rid of this stack of uuids and use only AllureReporter methods. @sseliverstov what do you say?

Reply:
@ehborisov yes, i think I thought about it when I wrote this code, but I tried not to affect the code of commons. I think, now it can already be rewritten

Reply:
I added use get_last_item where possible. I can't refuse use self.stack because step results does not have the uuid attribute now

Reply:
ok, yeah, I apparently thought steps have uuids. Lets keep it.